zerver/middleware.py

```diff
@@ -250,6 +252,10 @@ def process_request(self, request: HttpRequest) -> None:
             # Avoid re-initializing request._log_data if it's already there.
             return
 
+        def process_user_agent() -> None:
+            request.client_name = get_client_name(request, is_browser_view=True)
```

Comment:
It is not correct to hardcode `is_browser_view=True`.  

I think what we need to do to resolve this is to refactor the `is_browser_view` logic to be done in `process_client`, rather than `get_client_name`.

Reply:
OK, I figured out how to do this in a reasonable way; I've merged a series of prep commits that take care of this problem by removing that argument.  See the series ending with ccf63ac66bdaa221c8859f711f8e7d53a834fc58.

Reply:
Thanks for cleaning up get_client_name. I have accordingly rebased my commits and now the codeflow seems smoother.